src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1626,11 +1627,37 @@ void Terminal::PreviewText(std::wstring_view input)
         // pad it out
         preview.insert(originalSize, expectedLenTillEnd - originalSize, L' ');
     }
-    snippetPreview.text = til::visualize_nonspace_control_codes(preview);
+
     // Build our composition data
+    // The text is just the trimmed command, with the spaces at the end.
+    snippetPreview.text = til::visualize_nonspace_control_codes(preview);
+
+    // The attributes depend on the $profile:experimental.rainbowSuggestions setting:.
     const auto len = snippetPreview.text.size();
     snippetPreview.attributes.clear();
-    snippetPreview.attributes.emplace_back(len, previewAttrs);
+
+    if (_rainbowSuggestions)
+    {
+        // Let's do something fun.
+
+        // Use the actual text length for the number of steps, not including the
+        // trailing spaces.
+        const float increment = 1.0f / originalSize;

```

Comment:
Doesn't this and the loop below result in us going way over 0.9̅9 f when there is a lot of padding, because the step size is based on the shorter string but the loop on R1646 is based on the longer string?

The comment for `from_hue` suggests that `[0, 1)` is the valid range.

Reply:
I suppose yea, though, that only applies to the spaces at the end, so you can't actually see that they're **R E D**